### PR TITLE
Memory exec integration

### PR DESCRIPTION
**Pull request by mmamayka**
Implements LD/ST instructions simulation (need to be tested)

### CODE REVIEW

File: include/besm-666/exec/executor.hpp [hunk 1]
```diff
@@ -46,17 +47,17 @@ class Executor {
...
-    void exec_LHU(Instruction const instr) {}
-    void exec_LWU(Instruction const instr) {}
-    void exec_SB(Instruction const instr) {}
-    void exec_SH(Instruction const instr) {}
-    void exec_SW(Instruction const instr) {}
-    void exec_SD(Instruction const instr) {}
+    void exec_LB(Instruction const instr);
+    void exec_LH(Instruction const instr);
+    void exec_LW(Instruction const instr);
+    void exec_LD(Instruction const instr);
```
Comment (levBagryansky) [inline include/besm-666/exec/executor.hpp:53]: @mmamayka please make `Instruction`s not `const` in all these methods here. It does not make sense:
Parameter 'instr' is const-qualified in the function declaration; const-qualification of parameters only has an effect in function definitions

File: include/besm-666/exec/executor.hpp [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
 #include "besm-666/instruction.hpp"
+#include "besm-666/memory/mmu.hpp"
 #include "besm-666/rv-instruction-op.hpp"
 
 namespace besm::exec {
 
 class Executor {
 public:
-    Executor() = default;
+    Executor(mem::MMU::SPtr mmu) : mmu_(mmu) {}
```
Comment (levBagryansky) [inline include/besm-666/exec/executor.hpp:12]: @mmamayka please make it 
1) `explicit Executor`
2)  `: mmu_(std::move(mmu)) {}`

File: include/besm-666/exec/executor.hpp [hunk 3]
```diff
@@ -46,17 +47,17 @@ class Executor {
     void exec_BGEU(Instruction const instr);
 
     // Will be implemented after memory system release
```
Comment (c71n93) [inline include/besm-666/exec/executor.hpp:49]: remove coment

File: include/besm-666/memory/mmu.hpp [hunk 4]
```diff
@@ -23,6 +25,8 @@ class MMU : public INonCopyable {
     void storeDWord(RV64Ptr address, RV64DWord value);
 
 private:
+    explicit MMU(PhysMem::SPtr pMem) : pMem_(pMem) {}
```
Comment (levBagryansky) [inline include/besm-666/memory/mmu.hpp:28]: @mmamayka What do you think about `explicit MMU(PhysMem::SPtr pMem) : pMem_(std::move(pMem)) {}`

File: src/exec/executor.cpp [hunk 5]
```diff
@@ -487,6 +487,109 @@ void Executor::exec_BGEU(Instruction const instr) {
     }
 }
 
+void Executor::exec_LB(Instruction const instr) {
+    RV64UDWord address = gprf_.read(instr.rs1) +
+                         util::SignExtend<RV64UDWord, 12>(instr.immidiate);
+    RV64UChar value = mmu_->loadByte(address);
+    RV64UDWord extendedValue = util::SignExtend<RV64UDWord, 8>(value);
```
Comment (c71n93) [inline src/exec/executor.cpp:494]: use static_cast

File: src/exec/executor.cpp [hunk 6]
```diff
@@ -487,6 +487,109 @@ void Executor::exec_BGEU(Instruction const instr) {
...
+
+    gprf_.write(instr.rd, extendedValue);
+
+    this->nextPC();
+}
+void Executor::exec_LH(Instruction const instr) {
+    RV64UDWord address = gprf_.read(instr.rs1) +
+                         util::SignExtend<RV64UDWord, 12>(instr.immidiate);
+    RV64UHWord value = mmu_->loadHWord(address);
+    RV64UDWord extendedValue = util::SignExtend<RV64UDWord, 16>(value);
```
Comment (c71n93) [inline src/exec/executor.cpp:504]: use static_cast

File: src/exec/executor.cpp [hunk 7]
```diff
@@ -487,6 +487,109 @@ void Executor::exec_BGEU(Instruction const instr) {
...
+
+    gprf_.write(instr.rd, extendedValue);
+
+    this->nextPC();
+}
+void Executor::exec_LW(Instruction const instr) {
+    RV64UDWord address = gprf_.read(instr.rs1) +
+                         util::SignExtend<RV64UDWord, 12>(instr.immidiate);
+    RV64UWord value = mmu_->loadWord(address);
+    RV64UDWord extendedValue = util::SignExtend<RV64UDWord, 32>(value);
```
Comment (c71n93) [inline src/exec/executor.cpp:514]: use static_cast

**Review by levBagryansky — CHANGES_REQUESTED** [review]
@mmamayka Looks good but there are several ways to improve it

File: include/besm-666/exec/executor.hpp (hunk 2 above)
Comment (levBagryansky) [reply include/besm-666/exec/executor.hpp:12]: @mmamayka 1) up to you

File: include/besm-666/exec/executor.hpp (hunk 1 above)
Comment (mmamayka) [reply include/besm-666/exec/executor.hpp:53]: You have requested constantness of this parameters previously..... -__-

File: include/besm-666/exec/executor.hpp (hunk 2 above)
Comment (mmamayka) [reply include/besm-666/exec/executor.hpp:12]: What reason to use `std::move` in this context?

File: include/besm-666/exec/executor.hpp (hunk 3 above)
Comment (mmamayka) [reply include/besm-666/exec/executor.hpp:49]: Done

File: src/exec/executor.cpp (hunk 5 above)
Comment (mmamayka) [reply src/exec/executor.cpp:494]: Done

File: src/exec/executor.cpp (hunk 6 above)
Comment (mmamayka) [reply src/exec/executor.cpp:504]: Done

File: src/exec/executor.cpp (hunk 7 above)
Comment (mmamayka) [reply src/exec/executor.cpp:514]: Done

File: include/besm-666/exec/executor.hpp (hunk 2 above)
Comment (mmamayka) [reply include/besm-666/exec/executor.hpp]: DONE!

File: include/besm-666/memory/mmu.hpp (hunk 4 above)
Comment (mmamayka) [reply include/besm-666/memory/mmu.hpp]: Done!

File: include/besm-666/exec/executor.hpp (hunk 1 above)
Comment (mmamayka) [reply include/besm-666/exec/executor.hpp:52]: I would ignore this to save definition & declaration equal

**Review by levBagryansky — APPROVED** [review]

**Review by c71n93 — APPROVED** [review]

**Pull request merged**